en/middleware/xrce_dds.md

```diff
@@ -161,12 +163,18 @@ For example, the PX4 simulator runs the XRCE-DDS client over UDP on port 8888, s
 MicroXRCEAgent udp4 -p 8888
 ```
 
-On RaspberryPi, if you're using a serial link and connected to the IO pins, you might connect using this command:
+When working with real hardware, the setup depends on the hardware, OS, and channel.
+For example, if you're using the RasPi `UART0` serial port, you might connect using this command (based on the information in [Raspberry Pi Documentation > Configuring UARTS](https://www.raspberrypi.com/documentation/computers/configuration.html#configuring-uarts)):
 
 ```sh
 sudo MicroXRCEAgent serial --dev /dev/AMA0 -b 921600
 ```
 
+:::note
+For more information about setting up communications channels see [Pixhawk + Companion Setup > Serial Port setup](../companion_computer/pixhawk_companion.md#serial-port-setup), and sub-documents.
+:::
```

Comment:
@beniaminopozzan @ViniciusAbrao This generalizes the info about agent startup, making it clear that "it depends", linking to the RaPi configuring UARTs to help people understand a bit more about this case.
I also link, as suggested, to the companion computers page. That has some general Linux setup.

@ViniciusAbrao What would be great is if you could add your setup information as a topic below https://docs.px4.io/main/en/companion_computer/pixhawk_companion.html 

If this can be general it would be "Raspberry Pi Companion". If it has to be specific to a particular RaPi then ""Raspberry Pi Xxxx Companion", where Xxxx is the version. I think it can be general, but we'd need notes to highlight things to check in a particular system.

What would be great is more than just this specific part of setup around ROS. RasPi is very popular, so would be good to have images of the hardware connection showing the lines. Explanation of how it is powered, and why done that way, etc.
The setup of the connection could then be part of this.

Does it make sense/is it possible?

Reply:
It definitively makes sense. The linked docs covers only for USB-to-serial adapter but most (not only Raspberry Pi) companion computers have hardware serial devices.

Reply:
Yes, it's a simplistic topic. In an ideal world you'd start with a doc that covers all connection scenarios for as much hardware and OS as possible. That requires someone with broad experience commit quite a lot of time. Getting support for that is hard.

So the way I approach these problems is to attempt to build out specific hardware scenarios, then generalize them.
So for example here if @ViniciusAbrao can create a RaPi Companion doc covering as much as possible about the connections then that might contain information about non-USB serial connections, which can then migrate to the parent doc. 

Thats my thinking, anyway.

From that we do things like potentially cross link to https://docs.px4.io/main/en/companion_computer/holybro_pixhawk_rpi_cm4_baseboard.html with more software setup instructions. The idea being that we make setup of any RaPi setup as seamless as possible. 

@ViniciusAbrao You onboard to create that doc? RaPi is a very popular option, so I think a "good start". One thing no one seems to cover on companion setups, at least in detail, is how to power the RaPi.

Reply:
Yes @hamishwillee ... I am going to create the doc covering the connections and the setup. I let you know when I finish it !!

Reply:
Thanks @ViniciusAbrao - awesome!!!! Any kind of ETA?

Reply:
> Thanks @ViniciusAbrao - awesome!!!! Any kind of ETA?

I am going to work on it until next week !!

Reply:
@hamishwillee I have finished the review. Please, find the PR at:

https://github.com/PX4/PX4-user_guide/pull/2318#issuecomment-1451006165

If it is not the best way to push it, please let me know. Also, maybe the English can be improved, so, feel free to edit with corrections or technical changes.

Reply:
Thanks @ViniciusAbrao - I've had a first look at it and asked some questions. Next week I'll look at your answers, restructure it to be more easy to extend later (for example if we add information about "how to connect via Ethernet") and link it in.

Would be good if you could push it as a branch other than main in the PR sometime in the next week - that makes editing much easier with my knowledge of git.

Reply:
Thanks @hamishwillee The new PR is done at:

https://github.com/PX4/PX4-user_guide/pull/2323

I answered your questions... please, let me know if some points still not clear.